client/state/sites/selectors.js

```diff
@@ -145,7 +170,7 @@ export function isJetpackMinimumVersion( state, siteId, version ) {
  * @return {?String}        Site slug
  */
 export function getSiteSlug( state, siteId ) {
-	const site = getSite( state, siteId );
+	const site = getRawSite( state, siteId );
```

Comment:
I agree with the idea of composing these properties from individual selectors. Developers still have the convenience of `getSite` containing computed properties, but can access and maintain those selectors independently (performant, easier to test).